rasa/server.py

```diff
@@ -644,6 +644,15 @@ def create_app(
 
     # Setup the Sanic-JWT extension
     if jwt_secret and jwt_method:
+        # `sanic-jwt` depends on having an available event loop when making the call to
+        # `Initialize`. If there is non, the server startup will fail with
+        # `There is no current event loop in thread 'MainThread'`.
+        try:
```

Comment:
If you have any idea how to write a meaningful tests, I'm happy to take them. It's quite difficult to reproduce outside of Docker as `pytest` etc will always set a loop in the background before running things

Reply:
Yeah, not sure about a unit test here, what you're saying makes sense. However the issue appeared in an integration test, right? And this fixes that integration test? In that case I think the integration tests are able to cover it, since the pytest runs in the GH workflow and not on any of the docker containers spawned by the docker compose setup

Reply:
it appears as soon as you start the Rasa Enterprise docker setup (e.g. via `docker-compose`). I've manually build the containers locally, and used them with a Rasa Enterprise docker-compose deployment and hence can confirm that this change fixes things.

What we'd need mid-term is proper production-like integration tests in Rasa Open Source.

Reply:
Ah right I see what you mean, I've been so focused in on doing Rasa X integration testing 😄 So it was the enterprise integration tests that failed (which makes sense as to how this made it this far without us noticing

> What we'd need mid-term is proper production-like integration tests in Rasa Open Source.

Definitely. I think that we'll have a QA person more focused on open source in the upcoming months, and I think it'd be a great initiative to move the endpoints tests to a proper server setup (e.g. `test_server.py`) cc @mprazz for the roadmap 👀